libs/books/feature/src/lib/book-search/book-search.component.ts

```diff
@@ -62,7 +96,19 @@ export class BookSearchComponent implements OnInit,OnDestroy {
     }
   }
 
+
+  openSnackBar(message: string, action: string) {
+    const snackbar=this.snackBar.open(message, action,{
```

Comment:
Should be able to handle this without `this.item`.

Also, try moving the snackbar service to the books effects class instead. The component doesn't really need to be changed.

Reply:
moved the snackbar service to effects